# Perkenalan
Nama : Luthfi Fadhlillah
(FTDS Batch 009)

# Membuat Data Baru Kasus Covid-19
Instruksi: 
1. Jumlah kasus kumulativ tertinggi pada tanggal 20 Januari 2022
2. Puncak tertinggi kasus harian di Indonesia

# Import Pustaka
Melakukan import library/pustaka yang akan digunakan

In [5]:
import pandas as pd
import numpy as np

# Data Loading
Membuka data yang akan digunakan

In [6]:
data = pd.read_csv('https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_confirmed_global.csv')

In [7]:
data

Province/State        Country/Region        Lat        Long  1/22/20  \
0              NaN           Afghanistan  33.939110   67.709953        0   
1              NaN               Albania  41.153300   20.168300        0   
2              NaN               Algeria  28.033900    1.659600        0   
3              NaN               Andorra  42.506300    1.521800        0   
4              NaN                Angola -11.202700   17.873900        0   
..             ...                   ...        ...         ...      ...   
279            NaN    West Bank and Gaza  31.952200   35.233200        0   
280            NaN  Winter Olympics 2022  39.904200  116.407400        0   
281            NaN                 Yemen  15.552727   48.516388        0   
282            NaN                Zambia -13.133897   27.849332        0   
283            NaN              Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  2/11/22  2/12/22  \
0          0        0        0        0        0  ...   169940   170152   
1          0        0        0        0        0  ...   267020   267551   
2          0        0        0        0        0  ...   260723   261226   
3          0        0        0        0        0  ...    37140    37140   
4          0        0        0        0        0  ...    98501    98514   
..       ...      ...      ...      ...      ...  ...      ...      ...   
279        0        0        0        0        0  ...   600128   600128   
280        0        0        0        0        0  ...      490      498   
281        0        0        0        0        0  ...    11659    11664   
282        0        0        0        0        0  ...   309013   309293   
283        0        0        0        0        0  ...   231214   231299   

     2/13/22  2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  2/20/22  
0     170604   171246   171422   171519   171673   171857   171931   172205  
1     268008   268304   268491   268940   269301   269601   269904   270164  
2     261752   262165   262570   262994   263369   263685   263936   264054  
3      37140    37277    37361    37452    37522    37589    37589    37589  
4      98514    98514    98555    98568    98585    98605    98617    98638  
..       ...      ...      ...      ...      ...      ...      ...      ...  
279   600128   613846   618527   622175   625028   627956   627956   627956  
280      501      501      505      507      507      508      508      509  
281    11671    11679    11699    11707    11718    11731    11731    11736  
282   309502   309579   309870   310155   310474   310764   311016   311194  
283   231381   231603   231603   232213   232598   233030   233224   233352  

[284 rows x 765 columns]

# Data Exploration
Melakukan data eksplorasi pada datasheet secara sederhana.

In [8]:
data

Province/State        Country/Region        Lat        Long  1/22/20  \
0              NaN           Afghanistan  33.939110   67.709953        0   
1              NaN               Albania  41.153300   20.168300        0   
2              NaN               Algeria  28.033900    1.659600        0   
3              NaN               Andorra  42.506300    1.521800        0   
4              NaN                Angola -11.202700   17.873900        0   
..             ...                   ...        ...         ...      ...   
279            NaN    West Bank and Gaza  31.952200   35.233200        0   
280            NaN  Winter Olympics 2022  39.904200  116.407400        0   
281            NaN                 Yemen  15.552727   48.516388        0   
282            NaN                Zambia -13.133897   27.849332        0   
283            NaN              Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  2/11/22  2/12/22  \
0          0        0        0        0        0  ...   169940   170152   
1          0        0        0        0        0  ...   267020   267551   
2          0        0        0        0        0  ...   260723   261226   
3          0        0        0        0        0  ...    37140    37140   
4          0        0        0        0        0  ...    98501    98514   
..       ...      ...      ...      ...      ...  ...      ...      ...   
279        0        0        0        0        0  ...   600128   600128   
280        0        0        0        0        0  ...      490      498   
281        0        0        0        0        0  ...    11659    11664   
282        0        0        0        0        0  ...   309013   309293   
283        0        0        0        0        0  ...   231214   231299   

     2/13/22  2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  2/20/22  
0     170604   171246   171422   171519   171673   171857   171931   172205  
1     268008   268304   268491   268940   269301   269601   269904   270164  
2     261752   262165   262570   262994   263369   263685   263936   264054  
3      37140    37277    37361    37452    37522    37589    37589    37589  
4      98514    98514    98555    98568    98585    98605    98617    98638  
..       ...      ...      ...      ...      ...      ...      ...      ...  
279   600128   613846   618527   622175   625028   627956   627956   627956  
280      501      501      505      507      507      508      508      509  
281    11671    11679    11699    11707    11718    11731    11731    11736  
282   309502   309579   309870   310155   310474   310764   311016   311194  
283   231381   231603   231603   232213   232598   233030   233224   233352  

[284 rows x 765 columns]

In [9]:
print('Jumlah row', data.shape[0], '\nJumlah column', data.shape[1])

Jumlah row 284 
Jumlah column 765


Dapat dilihat bahwa data ini memiliki 284 baris dan 764 column

In [10]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 284 entries, 0 to 283
Columns: 765 entries, Province/State to 2/20/22
dtypes: float64(2), int64(761), object(2)
memory usage: 1.7+ MB


Data ini memiliki 2 column dengan tipe data float, 760 column dengan tipe data integer, dan 2 column dengan tipe data object

In [11]:
data.describe().T

count          mean           std       min          25%  \
Lat      282.0  2.010637e+01  2.584145e+01  -71.9499     4.643279   
Long     282.0  2.195872e+01  7.589337e+01 -178.1165   -22.036550   
1/22/20  284.0  1.961268e+00  2.640260e+01    0.0000     0.000000   
1/23/20  284.0  2.306338e+00  2.649984e+01    0.0000     0.000000   
1/24/20  284.0  3.313380e+00  3.299262e+01    0.0000     0.000000   
...        ...           ...           ...       ...          ...   
2/16/22  284.0  1.470893e+06  6.017744e+06    0.0000  7902.500000   
2/17/22  284.0  1.478046e+06  6.031356e+06    0.0000  7963.000000   
2/18/22  284.0  1.484923e+06  6.046653e+06    0.0000  7986.750000   
2/19/22  284.0  1.490334e+06  6.055395e+06    0.0000  7988.750000   
2/20/22  284.0  1.493938e+06  6.061073e+06    0.0000  7991.750000   

                  50%            75%           max  
Lat         21.607878      40.950592  7.170690e+01  
Long        20.921188      84.992575  1.780650e+02  
1/22/20      0.000000       0.000000  4.440000e+02  
1/23/20      0.000000       0.000000  4.440000e+02  
1/24/20      0.000000       0.000000  5.490000e+02  
...               ...            ...           ...  
2/16/22  63193.500000  736980.250000  7.817436e+07  
2/17/22  63193.500000  738947.500000  7.827570e+07  
2/18/22  63234.000000  740284.250000  7.842352e+07  
2/19/22  63528.500000  741032.000000  7.845708e+07  
2/20/22  63528.500000  742198.000000  7.847913e+07  

[763 rows x 8 columns]

In [12]:
data

Province/State        Country/Region        Lat        Long  1/22/20  \
0              NaN           Afghanistan  33.939110   67.709953        0   
1              NaN               Albania  41.153300   20.168300        0   
2              NaN               Algeria  28.033900    1.659600        0   
3              NaN               Andorra  42.506300    1.521800        0   
4              NaN                Angola -11.202700   17.873900        0   
..             ...                   ...        ...         ...      ...   
279            NaN    West Bank and Gaza  31.952200   35.233200        0   
280            NaN  Winter Olympics 2022  39.904200  116.407400        0   
281            NaN                 Yemen  15.552727   48.516388        0   
282            NaN                Zambia -13.133897   27.849332        0   
283            NaN              Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  2/11/22  2/12/22  \
0          0        0        0        0        0  ...   169940   170152   
1          0        0        0        0        0  ...   267020   267551   
2          0        0        0        0        0  ...   260723   261226   
3          0        0        0        0        0  ...    37140    37140   
4          0        0        0        0        0  ...    98501    98514   
..       ...      ...      ...      ...      ...  ...      ...      ...   
279        0        0        0        0        0  ...   600128   600128   
280        0        0        0        0        0  ...      490      498   
281        0        0        0        0        0  ...    11659    11664   
282        0        0        0        0        0  ...   309013   309293   
283        0        0        0        0        0  ...   231214   231299   

     2/13/22  2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  2/20/22  
0     170604   171246   171422   171519   171673   171857   171931   172205  
1     268008   268304   268491   268940   269301   269601   269904   270164  
2     261752   262165   262570   262994   263369   263685   263936   264054  
3      37140    37277    37361    37452    37522    37589    37589    37589  
4      98514    98514    98555    98568    98585    98605    98617    98638  
..       ...      ...      ...      ...      ...      ...      ...      ...  
279   600128   613846   618527   622175   625028   627956   627956   627956  
280      501      501      505      507      507      508      508      509  
281    11671    11679    11699    11707    11718    11731    11731    11736  
282   309502   309579   309870   310155   310474   310764   311016   311194  
283   231381   231603   231603   232213   232598   233030   233224   233352  

[284 rows x 765 columns]

- Berapa banyak data yang kasus kumulatif pada tanggal 20 Januari untuk negara-negara dengan latitude di bawah 0?

In [33]:
data[data['Lat'] < 0][['Country/Region', '1/20/22']].sort_values('1/20/22', ascending=False).head() #Defaultynya ascending=True dimana pandas akan menguruutkan nilai dari yang terbesar

Country/Region   1/20/22
31          Brazil  23595178
7        Argentina   7576335
149      Indonesia   4277644
236   South Africa   3572860
212           Peru   2780049

1. Negara dengan kasus kumulatif terbanyak pada tanggal 20 januari 2022 adalah brazil dengan total kasus 23 jutaan. Sedangkan indonesia diurutan ketiga dengan total kasus 4 jutaan yang artinya total kasus di Brazil lebih tinggi hampir 6 kali lipat daripada di Indonesia.

2. Buatlah fungsi untuk menghitung jumlah kasus harian Covid-19! (Hint: Kamu bisa menggunakan .diff() untuk menghitung perbedaan tiap baris data/menghitung jumlah kasus harian dan .transpose() untuk mengubah kolom menjadi baris atau sebaliknya)

In [40]:
data_new = data.drop(columns=['Province/State', 'Lat', 'Long'])
data_trans = data_new.set_index('Country/Region').transpose()
data_trans.head()

Country/Region  Afghanistan  Albania  Algeria  Andorra  Angola  Antarctica  \
1/22/20                   0        0        0        0       0           0   
1/23/20                   0        0        0        0       0           0   
1/24/20                   0        0        0        0       0           0   
1/25/20                   0        0        0        0       0           0   
1/26/20                   0        0        0        0       0           0   

Country/Region  Antigua and Barbuda  Argentina  Armenia  Australia  ...  \
1/22/20                           0          0        0          0  ...   
1/23/20                           0          0        0          0  ...   
1/24/20                           0          0        0          0  ...   
1/25/20                           0          0        0          0  ...   
1/26/20                           0          0        0          0  ...   

Country/Region  Uruguay  Uzbekistan  Vanuatu  Venezuela  Vietnam  \
1/22/20               0           0        0          0        0   
1/23/20               0           0        0          0        2   
1/24/20               0           0        0          0        2   
1/25/20               0           0        0          0        2   
1/26/20               0           0        0          0        2   

Country/Region  West Bank and Gaza  Winter Olympics 2022  Yemen  Zambia  \
1/22/20                          0                     0      0       0   
1/23/20                          0                     0      0       0   
1/24/20                          0                     0      0       0   
1/25/20                          0                     0      0       0   
1/26/20                          0                     0      0       0   

Country/Region  Zimbabwe  
1/22/20                0  
1/23/20                0  
1/24/20                0  
1/25/20                0  
1/26/20                0  

[5 rows x 284 columns]

In [41]:
data_trans.diff().dropna()

Country/Region  Afghanistan   Albania   Algeria  Andorra   Angola  Antarctica  \
1/23/20                 0.0       0.0       0.0      0.0      0.0         0.0   
1/24/20                 0.0       0.0       0.0      0.0      0.0         0.0   
1/25/20                 0.0       0.0       0.0      0.0      0.0         0.0   
1/26/20                 0.0       0.0       0.0      0.0      0.0         0.0   
1/27/20                 0.0       0.0       0.0      0.0      0.0         0.0   
...                     ...       ...       ...      ...      ...         ...   
2/17/22               154.0     361.0     375.0     70.0     17.0         0.0   
2/18/22               184.0     300.0     316.0     67.0     20.0         0.0   
2/19/22                74.0     303.0     251.0      0.0     12.0         0.0   
2/20/22               274.0     260.0     118.0      0.0     21.0         0.0   
Kasus harian      -172131.0 -269861.0 -263803.0 -37589.0 -98626.0       -11.0   

Country/Region  Antigua and Barbuda  Argentina   Armenia  Australia  ...  \
1/23/20                         0.0        0.0       0.0        0.0  ...   
1/24/20                         0.0        0.0       0.0        0.0  ...   
1/25/20                         0.0        0.0       0.0        0.0  ...   
1/26/20                         0.0        0.0       0.0        0.0  ...   
1/27/20                         0.0        0.0       0.0        0.0  ...   
...                             ...        ...       ...        ...  ...   
2/17/22                         0.0    16650.0    1723.0      522.0  ...   
2/18/22                         5.0    15389.0    1417.0      332.0  ...   
2/19/22                         8.0     7807.0    1469.0      551.0  ...   
2/20/22                         0.0     4450.0     700.0      435.0  ...   
Kasus harian                -7400.0 -8819697.0 -413995.0   -44535.0  ...   

Country/Region   Uruguay  Uzbekistan  Vanuatu  Venezuela    Vietnam  \
1/23/20              0.0         0.0      0.0        0.0        2.0   
1/24/20              0.0         0.0      0.0        0.0        0.0   
1/25/20              0.0         0.0      0.0        0.0        0.0   
1/26/20              0.0         0.0      0.0        0.0        0.0   
1/27/20              0.0         0.0      0.0        0.0        0.0   
...                  ...         ...      ...        ...        ...   
2/17/22           5517.0       241.0      0.0      926.0    36200.0   
2/18/22           7179.0       272.0      0.0      979.0    42439.0   
2/19/22           4355.0       191.0      4.0      606.0    54830.0   
2/20/22           3337.0       204.0      0.0        0.0    47200.0   
Kasus harian   -811349.0   -235076.0     -7.0  -509947.0 -2732663.0   

Country/Region  West Bank and Gaza  Winter Olympics 2022    Yemen    Zambia  \
1/23/20                        0.0                   0.0      0.0       0.0   
1/24/20                        0.0                   0.0      0.0       0.0   
1/25/20                        0.0                   0.0      0.0       0.0   
1/26/20                        0.0                   0.0      0.0       0.0   
1/27/20                        0.0                   0.0      0.0       0.0   
...                            ...                   ...      ...       ...   
2/17/22                     2853.0                   0.0     11.0     319.0   
2/18/22                     2928.0                   1.0     13.0     290.0   
2/19/22                        0.0                   0.0      0.0     252.0   
2/20/22                        0.0                   1.0      5.0     178.0   
Kasus harian             -627956.0                -509.0 -11736.0 -310942.0   

Country/Region  Zimbabwe  
1/23/20              0.0  
1/24/20              0.0  
1/25/20              0.0  
1/26/20              0.0  
1/27/20              0.0  
...                  ...  
2/17/22            385.0  
2/18/22            432.0  
2/19/22            194.0  
2/20/22            128.0  
Kasus harian   

In [44]:
def kasus_harian():
    data_new = data.drop(columns=['Province/State', 'Lat', 'Long'])
    data_trans = data_new.set_index('Country/Region').transpose()
    data_harian = data_trans.diff().dropna()
    return data_harian

In [47]:
kasus_harian()

Country/Region  Afghanistan   Albania   Algeria  Andorra   Angola  Antarctica  \
1/23/20                 0.0       0.0       0.0      0.0      0.0         0.0   
1/24/20                 0.0       0.0       0.0      0.0      0.0         0.0   
1/25/20                 0.0       0.0       0.0      0.0      0.0         0.0   
1/26/20                 0.0       0.0       0.0      0.0      0.0         0.0   
1/27/20                 0.0       0.0       0.0      0.0      0.0         0.0   
...                     ...       ...       ...      ...      ...         ...   
2/17/22               154.0     361.0     375.0     70.0     17.0         0.0   
2/18/22               184.0     300.0     316.0     67.0     20.0         0.0   
2/19/22                74.0     303.0     251.0      0.0     12.0         0.0   
2/20/22               274.0     260.0     118.0      0.0     21.0         0.0   
Kasus harian      -172131.0 -269861.0 -263803.0 -37589.0 -98626.0       -11.0   

Country/Region  Antigua and Barbuda  Argentina   Armenia  Australia  ...  \
1/23/20                         0.0        0.0       0.0        0.0  ...   
1/24/20                         0.0        0.0       0.0        0.0  ...   
1/25/20                         0.0        0.0       0.0        0.0  ...   
1/26/20                         0.0        0.0       0.0        0.0  ...   
1/27/20                         0.0        0.0       0.0        0.0  ...   
...                             ...        ...       ...        ...  ...   
2/17/22                         0.0    16650.0    1723.0      522.0  ...   
2/18/22                         5.0    15389.0    1417.0      332.0  ...   
2/19/22                         8.0     7807.0    1469.0      551.0  ...   
2/20/22                         0.0     4450.0     700.0      435.0  ...   
Kasus harian                -7400.0 -8819697.0 -413995.0   -44535.0  ...   

Country/Region   Uruguay  Uzbekistan  Vanuatu  Venezuela    Vietnam  \
1/23/20              0.0         0.0      0.0        0.0        2.0   
1/24/20              0.0         0.0      0.0        0.0        0.0   
1/25/20              0.0         0.0      0.0        0.0        0.0   
1/26/20              0.0         0.0      0.0        0.0        0.0   
1/27/20              0.0         0.0      0.0        0.0        0.0   
...                  ...         ...      ...        ...        ...   
2/17/22           5517.0       241.0      0.0      926.0    36200.0   
2/18/22           7179.0       272.0      0.0      979.0    42439.0   
2/19/22           4355.0       191.0      4.0      606.0    54830.0   
2/20/22           3337.0       204.0      0.0        0.0    47200.0   
Kasus harian   -811349.0   -235076.0     -7.0  -509947.0 -2732663.0   

Country/Region  West Bank and Gaza  Winter Olympics 2022    Yemen    Zambia  \
1/23/20                        0.0                   0.0      0.0       0.0   
1/24/20                        0.0                   0.0      0.0       0.0   
1/25/20                        0.0                   0.0      0.0       0.0   
1/26/20                        0.0                   0.0      0.0       0.0   
1/27/20                        0.0                   0.0      0.0       0.0   
...                            ...                   ...      ...       ...   
2/17/22                     2853.0                   0.0     11.0     319.0   
2/18/22                     2928.0                   1.0     13.0     290.0   
2/19/22                        0.0                   0.0      0.0     252.0   
2/20/22                        0.0                   1.0      5.0     178.0   
Kasus harian             -627956.0                -509.0 -11736.0 -310942.0   

Country/Region  Zimbabwe  
1/23/20              0.0  
1/24/20              0.0  
1/25/20              0.0  
1/26/20              0.0  
1/27/20              0.0  
...                  ...  
2/17/22            385.0  
2/18/22            432.0  
2/19/22            194.0  
2/20/22            128.0  
Kasus harian   

3. Pada tanggal berapa puncak tertinggi kasus harian Covid-19 di Indonesia? bandingkan dengan tanggal 20 Januari 2022, apakah mengalami penurunan atau kenaikan jumlah kasus?

In [48]:
data_kasus_harian = kasus_harian()

In [51]:
indo = data_kasus_harian['Indonesia']
puncak_indo = indo.max()
indo[indo==puncak_indo]

2/16/22    64718.0
Name: Indonesia, dtype: float64

In [52]:
indo.loc['1/20/22']

2116.0

Puncak kasus harian covid-19 di Indonesia di tanggal 16 Februari 2022 dengan jumlah kasus 64718, sedangkan di tanggal 20 Januari 2022, jumlah kasus hariannya adalah 2116, maka ada kenaikan jumlah kasus yang cukup signifikan.

4. Gunakan for loop dan conditional if untuk membuat sebuah list yang berisikan jumlah kasus harian Covid-19 dengan ketentuan hanya mengambil data Indonesia saja dan jumlah kasus di bawah 10000. Berapa jumlah hari dengan kasus di bawah 10000?

In [53]:
kasus_10000 = []
for baris in indo:
    if baris<10000:
        kasus_10000.append(baris)


In [55]:
print(kasus_10000)

[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 2.0, 0.0, 0.0, 0.0, 2.0, 0.0, 2.0, 13.0, 8.0, 7.0, 0.0, 35.0, 27.0, 21.0, 17.0, 38.0, 55.0, 84.0, 58.0, 81.0, 64.0, 65.0, 107.0, 104.0, 103.0, 153.0, 109.0, 130.0, 129.0, 114.0, 149.0, 113.0, 196.0, 106.0, 181.0, 218.0, 247.0, 218.0, 337.0, 219.0, 330.0, 399.0, 316.0, 282.0, 297.0, 380.0, 407.0, 325.0, 327.0, 185.0, 375.0, 283.0, 357.0, 436.0, 396.0, 275.0, 214.0, 415.0, 260.0, 347.0, 433.0, 292.0, 349.0, 395.0, 484.0, 367.0, 338.0, 336.0, 533.0, 387.0, 233.0, 484.0, 689.0, 568.0, 490.0, 529.0, 489.0, 496.0, 486.0, 693.0, 973.0, 634.0, 949.0, 526.0, 479.0, 415.0, 686.0, 687.0, 678.0, 557.0, 700.0, 467.0, 609.0, 684.0, 585.0, 703.0, 993.0, 672.0, 847.0, 1043.0, 1240.0, 979.0, 1111.0, 1014.0, 857.0, 1017.0, 1106.0, 1031.0, 1331.0, 1041.0, 1226.0, 862.0, 954.0, 1051.0, 1113.0, 1178.0, 1240.0, 138

In [56]:
print('Jumlah hari yang kasus hariannya dibawah 10000 untuk indonesia sebanyak', len(kasus_10000), 'hari')

Jumlah hari yang kasus hariannya dibawah 10000 untuk indonesia sebanyak 634 hari


# Kebawahnya Salah

In [37]:
data[['1/20/22', '1/21/22']].diff()

1/20/22    1/21/22
0          NaN        NaN
1      82209.0    84666.0
2     -11042.0   -11857.0
3    -198269.0  -199300.0
4      63019.0    62651.0
..         ...        ...
279 -1615459.0 -1630154.0
280  -479343.0  -480583.0
281    10507.0    10585.0
282   289464.0   290002.0
283   -72419.0   -72626.0

[284 rows x 2 columns]

In [14]:
data[data['Lat'] < 0][['1/20/22']].count()

1/20/22    56
dtype: int64

Ada 56 data negara-negara dengan latitude dibawah 0 untuk kasus kumulatif per tanggal 20 Januari 2022

# Answer
Menampilkan 5 data teratas untuk melihat gambaran dataset

In [15]:
data.head()

Province/State Country/Region       Lat       Long  1/22/20  1/23/20  \
0            NaN    Afghanistan  33.93911  67.709953        0        0   
1            NaN        Albania  41.15330  20.168300        0        0   
2            NaN        Algeria  28.03390   1.659600        0        0   
3            NaN        Andorra  42.50630   1.521800        0        0   
4            NaN         Angola -11.20270  17.873900        0        0   

   1/24/20  1/25/20  1/26/20  1/27/20  ...  2/11/22  2/12/22  2/13/22  \
0        0        0        0        0  ...   169940   170152   170604   
1        0        0        0        0  ...   267020   267551   268008   
2        0        0        0        0  ...   260723   261226   261752   
3        0        0        0        0  ...    37140    37140    37140   
4        0        0        0        0  ...    98501    98514    98514   

   2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  2/20/22  
0   171246   171422   171519   171673   171857   171931   172205  
1   268304   268491   268940   269301   269601   269904   270164  
2   262165   262570   262994   263369   263685   263936   264054  
3    37277    37361    37452    37522    37589    37589    37589  
4    98514    98555    98568    98585    98605    98617    98638  

[5 rows x 765 columns]

In [16]:
data_baru = data[data['Lat'] < 0][['1/20/22']]

In [17]:
data_baru

1/20/22
4       95220
5          11
7     7576335
9       28472
10     919876
11       8219
12     298367
13      89043
14      23515
15     717692
16       1338
28     795087
30     243946
31   23595178
36      36868
58    1916522
94       7784
95      23244
96      84198
110    658045
116     68003
118     60509
121     47275
124     35189
125     14854
126    133617
130       454
132     45663
149   4277644
159    319379
160        37
168     31899
175     55827
176     83403
183     70094
191    221432
192    154897
199         2
200     15401
210     36446
211    516555
212   2780049
219    127227
223        13
229     34367
234        81
236   3572860
248     32393
250     19866
252         1
265        85
271         4
274    556163
276         7
282    299971
283    227552

In [18]:
data_baru['1/20/22'].max()

23595178

Data kasus tertinggi pada tanggal 20 Januari 2022 untuk negara dengan dibawah latitude 0 adalah sebesar 23595178

In [19]:
data[data['Lat'] < 0][['1/20/22', "Country/Region"]].max()

1/20/22           23595178
Country/Region    Zimbabwe
dtype: object

Dari data diatas, negara Zimbabwe merupakan negara dengan kasus kumulatif paling tinggi pada tanggal 20 Januari 2022 untuk negara negara dengan latitude dibawah 0.

In [20]:
data.loc[data['Country/Region']=='Indonesia',:]['1/20/22']

149    4277644
Name: 1/20/22, dtype: int64

Data Indonesia pada tanggal 20 januari 2022 untuk kasus kumulatif adalah sebesar 4277644

In [21]:
data.head()

Province/State Country/Region       Lat       Long  1/22/20  1/23/20  \
0            NaN    Afghanistan  33.93911  67.709953        0        0   
1            NaN        Albania  41.15330  20.168300        0        0   
2            NaN        Algeria  28.03390   1.659600        0        0   
3            NaN        Andorra  42.50630   1.521800        0        0   
4            NaN         Angola -11.20270  17.873900        0        0   

   1/24/20  1/25/20  1/26/20  1/27/20  ...  2/11/22  2/12/22  2/13/22  \
0        0        0        0        0  ...   169940   170152   170604   
1        0        0        0        0  ...   267020   267551   268008   
2        0        0        0        0  ...   260723   261226   261752   
3        0        0        0        0  ...    37140    37140    37140   
4        0        0        0        0  ...    98501    98514    98514   

   2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  2/20/22  
0   171246   171422   171519   171673   171857   171931   172205  
1   268304   268491   268940   269301   269601   269904   270164  
2   262165   262570   262994   263369   263685   263936   264054  
3    37277    37361    37452    37522    37589    37589    37589  
4    98514    98555    98568    98585    98605    98617    98638  

[5 rows x 765 columns]

In [22]:
def Kasus_harian():
    "Menghitung kasus harian covid-19"
    Kasus = hari_ini - hari_kemarin
    return Kasus

In [23]:
data['Kasus harian'] = ''

In [24]:
data

Province/State        Country/Region        Lat        Long  1/22/20  \
0              NaN           Afghanistan  33.939110   67.709953        0   
1              NaN               Albania  41.153300   20.168300        0   
2              NaN               Algeria  28.033900    1.659600        0   
3              NaN               Andorra  42.506300    1.521800        0   
4              NaN                Angola -11.202700   17.873900        0   
..             ...                   ...        ...         ...      ...   
279            NaN    West Bank and Gaza  31.952200   35.233200        0   
280            NaN  Winter Olympics 2022  39.904200  116.407400        0   
281            NaN                 Yemen  15.552727   48.516388        0   
282            NaN                Zambia -13.133897   27.849332        0   
283            NaN              Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  2/12/22  2/13/22  \
0          0        0        0        0        0  ...   170152   170604   
1          0        0        0        0        0  ...   267551   268008   
2          0        0        0        0        0  ...   261226   261752   
3          0        0        0        0        0  ...    37140    37140   
4          0        0        0        0        0  ...    98514    98514   
..       ...      ...      ...      ...      ...  ...      ...      ...   
279        0        0        0        0        0  ...   600128   600128   
280        0        0        0        0        0  ...      498      501   
281        0        0        0        0        0  ...    11664    11671   
282        0        0        0        0        0  ...   309293   309502   
283        0        0        0        0        0  ...   231299   231381   

     2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  2/20/22  \
0     171246   171422   171519   171673   171857   171931   172205   
1     268304   268491   268940   269301   269601   269904   270164   
2     262165   262570   262994   263369   263685   263936   264054   
3      37277    37361    37452    37522    37589    37589    37589   
4      98514    98555    98568    98585    98605    98617    98638   
..       ...      ...      ...      ...      ...      ...      ...   
279   613846   618527   622175   625028   627956   627956   627956   
280      501      505      507      507      508      508      509   
281    11679    11699    11707    11718    11731    11731    11736   
282   309579   309870   310155   310474   310764   311016   311194   
283   231603   231603   232213   232598   233030   233224   233352   

     Kasus harian  
0                  
1                  
2                  
3                  
4                  
..            ...  
279                
280                
281                
282                
283                

[284 rows x 766 columns]

In [25]:
data['Kasus harian'] = data['2/19/22'] - data['2/18/22']

In [26]:
data['Kasus harian']

0       74
1      303
2      251
3        0
4       12
      ... 
279      0
280      0
281      0
282    252
283    194
Name: Kasus harian, Length: 284, dtype: int64

Data kasus harian pertanggal 20 Februari 2022

In [27]:
data

Province/State        Country/Region        Lat        Long  1/22/20  \
0              NaN           Afghanistan  33.939110   67.709953        0   
1              NaN               Albania  41.153300   20.168300        0   
2              NaN               Algeria  28.033900    1.659600        0   
3              NaN               Andorra  42.506300    1.521800        0   
4              NaN                Angola -11.202700   17.873900        0   
..             ...                   ...        ...         ...      ...   
279            NaN    West Bank and Gaza  31.952200   35.233200        0   
280            NaN  Winter Olympics 2022  39.904200  116.407400        0   
281            NaN                 Yemen  15.552727   48.516388        0   
282            NaN                Zambia -13.133897   27.849332        0   
283            NaN              Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  2/12/22  2/13/22  \
0          0        0        0        0        0  ...   170152   170604   
1          0        0        0        0        0  ...   267551   268008   
2          0        0        0        0        0  ...   261226   261752   
3          0        0        0        0        0  ...    37140    37140   
4          0        0        0        0        0  ...    98514    98514   
..       ...      ...      ...      ...      ...  ...      ...      ...   
279        0        0        0        0        0  ...   600128   600128   
280        0        0        0        0        0  ...      498      501   
281        0        0        0        0        0  ...    11664    11671   
282        0        0        0        0        0  ...   309293   309502   
283        0        0        0        0        0  ...   231299   231381   

     2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  2/20/22  \
0     171246   171422   171519   171673   171857   171931   172205   
1     268304   268491   268940   269301   269601   269904   270164   
2     262165   262570   262994   263369   263685   263936   264054   
3      37277    37361    37452    37522    37589    37589    37589   
4      98514    98555    98568    98585    98605    98617    98638   
..       ...      ...      ...      ...      ...      ...      ...   
279   613846   618527   622175   625028   627956   627956   627956   
280      501      505      507      507      508      508      509   
281    11679    11699    11707    11718    11731    11731    11736   
282   309579   309870   310155   310474   310764   311016   311194   
283   231603   231603   232213   232598   233030   233224   233352   

     Kasus harian  
0              74  
1             303  
2             251  
3               0  
4              12  
..            ...  
279             0  
280             0  
281             0  
282           252  
283           194  

[284 rows x 766 columns]

In [28]:
data.loc[data['Country/Region']=='Indonesia',:].max()

Province/State         None
Country/Region    Indonesia
Lat                 -0.7893
Long               113.9213
1/22/20                   0
                    ...    
2/17/22             5030002
2/18/22             5089637
2/19/22             5149021
2/20/22             4966046
Kasus harian          59384
Length: 766, dtype: object